data/ext/bio/FormalParameter.rdfa

```diff
@@ -0,0 +1,40 @@
+<div>
+  <!--
+  # Schema.org Bioschemas extension proposal
+  * | Project: https://bioschemas.org
+  * | MailingList: https://lists.w3.org/Archives/Public/public-bioschemas/
+  * | Issues:
+  -->
+
+  <div typeof="rdfs:Class" resource="http://schema.org/FormalParameter">
+      <link property="http://schema.org/isPartOf" href="http://bio.schema.org" />
+      <span class="h" property="rdfs:label">FormalParameter</span>
+      <span property="rdfs:comment">A formal parameter is a slot that may be satisfied when the workflow is run. It appears as an Input or Output of a Workflow</span>
+      <span>Subclass of: <a property="rdfs:subClassOf" href="http://schema.org/Intangible">Intangible</a></span>
+      <span>Source:  <a property="dc:source" href="http://bioschemas.org">Bioschemas</a></span>
+  </div>
+
+  <div typeof="rdf:Property" resource="http://schema.org/encodingFormat">
+    <span>Domain: <a property="http://schema.org/domainIncludes" href="http://schema.org/FormalParameter">FormalParameter</a></span>
+    <!-- from schema.rdfa
+    <span>Range: <a property="http://schema.org/rangeIncludes" href="http://schema.org/Text">Text</a></span>
+    <span>Range: <a property="http://schema.org/rangeIncludes" href="http://schema.org/Text">URL</a></span>
+     -->
+  </div>
+
+  <div typeof="rdf:Property" resource="http://schema.org/valueRequired">
+    <span property="rdfs:comment">Whether the input/output must be specified</span>
+    <span>Domain: <a property="http://schema.org/domainIncludes" href="http://schema.org/FormalParameter">FormalParameter</a></span>
+    <span>Range: <a property="http://schema.org/rangeIncludes" href="http://schema.org/Boolean">Boolean</a></span>
+    <span>Source:  <a property="dc:source" href="http://www.bioschemas.org/FormalParameter">FormalParameter</a></span>
+  </div>
+
+  <div typeof="rdf:Property" resource="http://schema.org/defaultValue">
+    <span>Domain: <a property="http://schema.org/domainIncludes" href="http://schema.org/FormalParameter">FormalParameter</a></span>
+    <!-- from schema.rdfa
+    <span class="h" property="rdfs:label">defaultValue</span>    
+    <span>Range: <a property="http://schema.org/rangeIncludes" href="http://schema.org/Thing">Thing</a></span>
+    -->
+  </div>
+
```

Comment:
What about a property specifying the nature/type of the input/output data? It could be a schema/Bioschemas type or EDAM (or both). Would it be useful for the Workflows case? I know it has been mentioned for the Software/Tools case

Reply:
better to do that in the [BioSchemas profile for FormalParameter](https://docs.google.com/spreadsheets/d/1VKntwxkgjvup6yfzaPZsBN3h5aKWU5nuTKGEds3GzeQ/edit#gid=1261485211) - where we say we should use `additionalType` to link to EDAM.

Reply:
Good point about the profile, but not via additionalType. The motivation of additionalType is allowing multiple classes in one of the serialization which does not natively support it (microdata, I think). That is why I thought of adding a new property here... although, if done via profiling, additionalType could actually work.

Reply:
A good thing about the looser `additionalType` here is that it is not the same RDF semantics as `@type` - in particular a `FormalParameter` with an additionalType of EDAM Genome Sequence is not actually a genome sequence, as it is a parameter _slot_ which would accept (or produce) values of that type. In the same way `additionalType` can type using identifiers to non-class instances, e.g. a `skos:Concept`.

As it's only in the profile this is straight forward.. if we want it in the `FormalParameter` type registration (this repo) then at best we would need to describe that pattern in free text for the class itself.

Say we added new properties instead, I don't know a better property that would work well on both input (`expectsValuesOfType` ?) and output (`producesValuesByType` ?)..

Reply:
Indeed additionalType makes sense here. @AlasdairGray any thoughts on this point?